tooling/minimize/src/bb.rs

```diff
@@ -194,9 +194,46 @@ impl<'cx, 'tcx> FnCtxt<'cx, 'tcx> {
 
                     if should_panic {
                         // FIXME trigger a panic instead of UB
-                        Terminator::Unreachable
+                        return Terminator::Unreachable;
                     } else {
-                        Terminator::Goto(self.bb_name_map[&target.unwrap()])
+                        return Terminator::Goto(self.bb_name_map[&target.unwrap()]);
+                    }
+                }
+                "unlikely" | "likely" => {
+                    // FIXME: translate this into an assignment statement + goto terminator
+                    // translate `unlikely`, `likely` intrinsic into functions that can be called
+                    let function_name = self.cx.get_fn_name(instance);
+                    if !self.functions.contains_key(function_name) {
+                        let ret = LocalName(Name::from_internal(0));
+                        let arg = LocalName(Name::from_internal(1));
+
+                        let b0_name = BbName(Name::from_internal(0));
+                        let assign = Statement::Assign {
+                            destination: PlaceExpr::Local(ret),
+                            source: ValueExpr::Load { source: GcCow::new(PlaceExpr::Local(arg)) },
+                        };
+                        let b0 = BasicBlock {
+                            statements: list![assign],
+                            terminator: Terminator::Return,
+                        };
+
+                        let mut blocks = Map::new();
+                        blocks.insert(b0_name, b0);
+
+                        let mut locals = Map::new();
+                        locals.insert(ret, <bool>::get_type());
+                        locals.insert(arg, <bool>::get_type());
+
+                        let function = Function {
+                            locals,
+                            args: list![arg],
+                            ret,
+                            blocks,
+                            start: b0_name,
+                            calling_convention: CallingConvention::Rust,
+                        };
+                        // can unwrap as checked if `function_name` already in functions map
```

Comment:
```suggestion
                        // We can unwrap because we already checked that `function_name` is not in the `functions` map.
```

Reply:
Proper sentences! :)

Reply:
Yes please. :) I had to read your comment 3 times before I understood it...